src/NUnitFramework/framework/Internal/Execution/WorkItem.cs

```diff
@@ -230,87 +230,66 @@ public virtual void Execute()
                 timeout = (int)Test.Properties.Get(PropertyNames.Timeout);
 
 #if SILVERLIGHT || NETCF
-            if (Test.RequiresThread || Test is TestMethod && timeout > 0)
+            if (Test is TestMethod)
                 RunTestOnOwnThread(timeout);
             else
                 RunTest();
 #elif PORTABLE
             RunTest();
 #else
-            ApartmentState currentApartment = Thread.CurrentThread.GetApartmentState();
+            currentApartment = Thread.CurrentThread.GetApartmentState();
 
-            if (Test.RequiresThread || Test is TestMethod && timeout > 0 || currentApartment != TargetApartment && TargetApartment != ApartmentState.Unknown)
+            if (Test is TestMethod)
```

Comment:
Two questions:

1. Given that we have been on a long-term path toward removing these "extra" threads, is there any way you can make this work without running the test method on its own thread? I can explain what i had planned in more detail if necessary, but in sum it involves eliminating SimpleWorkItemDispatcher and creating worker threads dynamically, as needed. This change pushes us a step back on that path. That said, if it's necessary, then we will have to live with it.

2. This appears to break the functionality when Apartment(STA) is placed on a class while running under the SimpleWorkItemDispatcher. In that case, the test won't be a TestMethod.